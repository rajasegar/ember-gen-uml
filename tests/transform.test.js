/* globals test expect */
const transform = require('../src/transform');

test('should generate transform public property properly', function() {
  const input = `
export default class MyComponent extends Component {
  prop1 = true;
}`;

  const uml = transform(input);

  expect(uml).toMatchSnapshot();
});

test('should generate transform multiple properties properly', function() {
  const input = `
export default class MyComponent extends Component {
  prop1 =  true;
  prop2 = 'hello';
}`;

  const uml = transform(input);

  expect(uml).toMatchSnapshot();
});

test('should generate transform public method properly', function() {
  const input = `
export default class MyComponent extends Component {
  method1() {
  }
}`;

  const uml = transform(input);
  expect(uml).toMatchSnapshot();
});

test('should generate transform service properties  properly', function() {
  const input = `
export default class MyComponent extends Component {
  @service marketplace;
  @service overlay;
}`;

  const uml = transform(input);
  expect(uml).toMatchSnapshot();
});

test('should generate model uml properly', function() {
  const input = `
export default class PersonModel extends Model {
  @attr title;
  @attr name;
  @attr birthday;
}`;

  const uml = transform(input);
  expect(uml).toMatchSnapshot();
});
