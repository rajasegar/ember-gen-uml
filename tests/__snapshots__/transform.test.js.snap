// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate model uml properly 1`] = `
"
@startuml
Model <|-- PersonModel

class PersonModel {
  +title
  +name
  +birthday
}

@enduml"
`;

exports[`should generate transform multiple properties properly 1`] = `
"
@startuml
Component <|-- MyComponent

class MyComponent {
  +prop1
  +prop2
}

@enduml"
`;

exports[`should generate transform public method properly 1`] = `
"
@startuml
Component <|-- MyComponent

class MyComponent {
  +method1()
}

@enduml"
`;

exports[`should generate transform public property properly 1`] = `
"
@startuml
Component <|-- MyComponent

class MyComponent {
  +prop1
}

@enduml"
`;

exports[`should generate transform service properties  properly 1`] = `
"
@startuml
Component <|-- MyComponent
class marketplace << (S, #FF7700) >>
MyComponent ..> marketplace : service
class overlay << (S, #FF7700) >>
MyComponent ..> overlay : service
class MyComponent {
  
}

@enduml"
`;
